com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -352,7 +352,47 @@ protected NetworkBehaviour GetNetworkBehaviour(ushort behaviourId)
         /// <summary>
         /// Gets the ClientId that owns the NetworkObject
         /// </summary>
-        public ulong OwnerClientId => NetworkObject.OwnerClientId;
+        public ulong OwnerClientId { get; internal set; }
+
+        /// <summary>
+        /// Updates properties with network session related
+        /// dependencies such as a NetworkObject's spawned
+        /// state or NetworkManager's session state.
+        /// </summary>
+        internal void UpdateNetworkProperties()
```

Comment:
Did you consider splitting this into multiple functions?  For instance, I don't think you need to re-evaluate NetworkBehaviourId when you change ownership or or despawn.  Then those functions would be more clear about what state updates are tied to which events.  Also, while probably not a major perf thing, you can avoid code that never needs to run again (e.g. IsHost/IsClient/IsServer updates)

Reply:
🤣 
Actually I had broken this up originally (see Fatih's suggested changes above) into two methods where the UpdateNetworkProperties was only invoked during spawn and despawn but OnOwnershipUpdate was invoked when ownership changed and during spawn and despawn.

Really, the total processing time incurred by setting everything during spawn, despawn, and ownership change is negligible relative to the way things were originally.